packages/server/graphql/mutations/helpers/generateGroups.ts

```diff
@@ -0,0 +1,28 @@
+import getRethink from '../../../database/rethinkDriver'
+import Reflection from '../../../database/types/Reflection'
+import OpenAIServerManager from '../../../utils/OpenAIServerManager'
+import {DataLoaderWorker} from '../../graphql'
+
+const generateGroups = async (
+  reflections: Reflection[],
+  teamId: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const {meetingId} = reflections[0]!
+  const team = await dataLoader.get('teams').loadNonNull(teamId)
+  const organization = await dataLoader.get('organizations').load(team.orgId)
+  const {featureFlags} = organization
+  const hasSuggestGroupsFlag = featureFlags?.includes('suggestGroups')
+  if (!hasSuggestGroupsFlag) return
+  const r = await getRethink()
+  const groupReflectionsInput = reflections.map((reflection) => reflection.plaintextContent)
+  const manager = new OpenAIServerManager()
+  const groupedReflectionsJSON = await manager.groupReflections(groupReflectionsInput)
+  if (!groupedReflectionsJSON) {
+    console.warn('ChatGPT was unable to group the reflections')
+    return
+  }
+  await r.table('NewMeeting').get(meetingId).update({groupedReflectionsJSON}).run()
```

Comment:
-1 I don't think we should store the groups on the meeting, but if we did, then it should be as an object directly, not as an unparsed JSON string.

Reply:
Because `NewMeeting` is still in rethink, you can just store the object directly in a field on `NewMeeting`, no reason to add another relation.
I think even in Postgres in the future this would be a good candidate for just a JSON field.

Reply:
The JSON includes both the new reflection group title and the text of the reflections in the group, so we couldn't add just an `autogroupReflectionsIds` field to the `Meeting` object if that's what you had in mind?

Here's the format of the JSON:

```
      [
        {
          "The Retreat": [
            "The retreat went well",
            "I liked that the length of the retreat",
            "I enjoyed the hotel"
          ]
        },
        {
          "Deadlines": [
            "Feeling overworked",
            "The project might not be ready in time",
            "The deadline is really tight"
          ]
        }
      ]
```

Reply:
What I had in mind is rather a format like
```
[
  {
    groupTitle: "The Retreat",
    reflectionIds: [12,23,45]
  },
  {
    groupTitle: "Deadlines",
    reflectionIds: [67,89,0]
  }
]
```
So do the processing and then store the object. That would make most sense to me, so it's easy to digest the data without knowing it's coming from openai